elisp/guix-deriviation.el

```diff
@@ -0,0 +1,89 @@
+;;; guix-deriviation.el --- Guix deriviation mode
+
+;; Copyright © 2017 Oleg Pykhalov <go.wigust@gmail.com>
+;; Copyright © 2017 Alex Kost <alezost@gmail.com>
+
+;; This file is part of Emacs-Guix.
+
+;; Emacs-Guix is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; Emacs-Guix is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with Emacs-Guix.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; This package provides minor-mode for prettifying Guix deriviation files.
+
+;;; Code:
+
+(require 'guix-utils)
+
+(defgroup guix-deriviation nil
+  "Prettify Guix deriviation files."
+  :prefix "guix-deriviation-file-"
+  :group 'guix
+  :group 'convenience)
+
+(defcustom guix-deriviation-file-regexp-group 1
+  "Regexp group in `guix-prettify-regexp' for prettifying."
+  :type 'integer
+  :group 'guix-deriviation)
+
+(defcustom guix-deriviation-file-regexp
+  (rx (group (and "/gnu" "/store")
+             ;; Hash-parts do not include "e", "o", "u" and "t".  See base32Chars
+             ;; at <https://github.com/NixOS/nix/blob/master/src/libutil/hash.cc>
+             "/" (= 32 (any "0-9" "a-d" "f-n" "p-s" "v-z"))
+             (* (any "-" "." "0-9" "a-z")) ".drv"))
+  "Regexp matching Guix deriviations for clicking."
+  :type 'regexp
+  :group 'guix-deriviation)
+
+(define-button-type 'guix-deriviation-file-find
+  'follow-link t
+  'action      #'guix-deriviation-file-find-button)
+
+(defun guix-deriviation-file-find-button (button)
+  "Guix find file BUTTON."
+  (guix-find-file (buffer-substring (button-start button)
+                                    (button-end button))))
+
+(defun guix-make-deriviation-button ()
```

Comment:
Since it creates multiple buttons potentially, I think using
`buttons` in the name would be more appropriate.  Also it is
better to begin all symbol names with `guix-derivation-`, so I
would call it `guix-derivation-make-buttons`.